flink/src/main/java/org/apache/iceberg/flink/writer/BaseTaskWriter.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.writer;
+
+import java.io.IOException;
+import java.util.List;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DataFiles;
+import org.apache.iceberg.Metrics;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.encryption.EncryptedOutputFile;
+import org.apache.iceberg.io.FileAppender;
+
+abstract class BaseTaskWriter<T> implements TaskWriter<T> {
+
+  protected DataFile closeFileAppender(FileAppender<T> fileAppender, EncryptedOutputFile currentFile,
+                                       PartitionSpec spec, StructLike partitionKey) throws IOException {
+    // Close the file appender firstly.
+    fileAppender.close();
+
+    // metrics are only valid after the appender is closed.
+    Metrics metrics = fileAppender.metrics();
+    long fileSizeInBytes = fileAppender.length();
+    List<Long> splitOffsets = fileAppender.splitOffsets();
+
+    // Construct the DataFile and add it into the completeDataFiles.
```

Comment:
Looks like this comment is no longer correct because `completeDataFiles` is private in the subclasses.